src/main/resources/io/viash/platforms/nextflow/WorkflowHelper.nf

```diff
@@ -603,10 +606,467 @@ def paramsToList(params, config) {
+    isInputFile = inputFileSetttings.find({it.plainName == parName})
+    if (isInputFile) {
+      if (parValue instanceof List) {
+        parValue = parValue.collect({path -> 
+          path !instanceof String ? path : file(getChild(relativeTo, path))
+        })
+      } else {
+        parValue = parValue !instanceof String ? path : file(getChild(relativeTo, parValue))
+      }
+    }
+    [parName, parValue]
+  }
+}
+
+/**
+ * Parse nextflow parameters based on settings defined in a viash config 
+ * and return a nextflow channel.
+ *
+ * @param params Input parameters from nextflow.
+ * @param multiArgumentsSettings A List of the viash configuration argument entries 
+ *                               that have the property 'multiple: true'.
+ * @param inputfileArguments A List of the viash configuration argument entries 
+ *                           that specify the input files.
+ *
+ * @return A list of parameter sets that were parsed from the 'param_list' argument value.
+ */
+private List<Tuple2<String, Map>> _parseParamListArguments(Map params, 
+  List<Map> multiArguments, List<Map> inputfileArguments){
+  // first try to guess the format (if not set in params)
+  def paramListFormat = _guessParamListFormat(params)
+
+  // get the correct parser function for the detected params_list format
+  def paramListParsers = [ 
+    "csv": {[it, readCsv(it)]},
+    "json": {[it, readJson(it)]},
+    "yaml": {[it, readYaml(it)]},
+    "yaml_blob": {[null, readYamlBlob(it)]},
+    "asis": {[null, it]},
+    "none": {[null, [[:]]]}
+  ]
+  assert paramListParsers.containsKey(paramListFormat):
+    "Format of provided --param_list not recognised.\n" +
+    "You can use '--param_list_format' to manually specify the format.\n" +
+    "Found: '$paramListFormat'. Expected: one of 'csv', 'json', "+
+    "'yaml', 'yaml_blob', 'asis' or 'none'"
+  def paramListParser = paramListParsers.get(paramListFormat)
+
+  // fetch multi param inputs
+  def paramListOut = paramListParser(params.containsKey("param_list") ? params.param_list : "")
+  // multiFile is null if the value passed to param_list was not a file (e.g a blob)
+  // If the value was indeed a file, multiFile contains the location that file (used later).
+  def paramListFile = paramListOut[0]
+  def paramSets = paramListOut[1] // these are the actual parameters from reading the blob/file
+
+  // data checks
+  assert paramSets instanceof List: "--param_list should contain a list of maps"
+  for (value in paramSets) {
+    assert value instanceof Map: "--param_list should contain a list of maps"
+  }
+
+  // Reformat from List<Map> to List<Tuple2<String, Map>> by adding the ID as first element of a Tuple2
+  paramSets = paramSets.collect({ paramValues ->
+    [paramValues.get("id", null), paramValues.findAll{it.key != 'id'}]
+  })
+  // Split parameters with 'multiple: true'
+  paramSets = paramSets.collect({ paramSet ->
+    def id = paramSet[0]
+    def paramValues = paramSet[1]
```

Comment:
Can use `paramSets = paramSets.collect({ id, data ->` notation since you know exactly what will be in `paramSets` here